app/services/platforms/youtube.ts

```diff
@@ -10,16 +10,82 @@ import {
 import { HostsService } from '../hosts';
 import { SettingsService } from '../settings';
 import { Inject } from '../core/injector';
-import { authorizedHeaders } from '../../util/requests';
+import { authorizedHeaders, handleResponse } from '../../util/requests';
 import { UserService } from '../user';
 import { $t } from 'services/i18n';
-import { handlePlatformResponse, platformAuthorizedRequest, platformRequest } from './utils';
+import { platformAuthorizedRequest, platformRequest } from './utils';
 
 interface IYoutubeServiceState {
   liveStreamingEnabled: boolean;
   liveStreamId: string;
   scheduledStartTime: string;
-  channelInfo: IChannelInfo;
+  liveBroadcast: IYoutubeLiveBroadcast;
+}
+
+/**
+ * Represents an API response with a paginated collection
+ */
+interface IYoutubeCollection<T> {
+  items: T[];
+  pageInfo: { totalResults: number; resultsPerPage: number };
+}
+
+/**
+ * A liveBroadcast resource represents an event that will be streamed, via live video, on YouTube.
+ * For the full set of available fields:
+ * @see https://google-developers.appspot.com/youtube/v3/live/docs/liveBroadcasts
+ */
+interface IYoutubeLiveBroadcast {
+  id: string;
+  contentDetails: {
+    boundStreamId: string;
+    enableAutoStart: boolean;
+  };
+  snippet: {
+    title: string;
+    description: string;
+    scheduledStartTime: string;
+    isDefaultBroadcast: boolean;
+    liveChatId: string;
+  };
+  status: {
+    lifeCycleStatus:
+      | 'complete'
+      | 'created'
+      | 'live'
+      | 'liveStarting'
+      | 'ready'
+      | 'revoked'
+      | 'testStarting'
+      | 'testing';
+    privacyStatus: 'private' | 'public' | 'unlisted';
+    recordingStatus: 'notRecording' | 'recorded' | 'recording';
+  };
+}
+
+/**
+ * A liveStream resource contains information about the video stream that you are transmitting to YouTube.
+ * The stream provides the content that will be broadcast to YouTube users. Once created,
+ * a liveStream resource can be bound to one or more liveBroadcast resources.
+ * @see https://google-developers.appspot.com/youtube/v3/live/docs/liveStreams
+ */
+interface IYoutubeLiveStream {
+  id: string;
+  snippet: {
+    isDefaultStream: boolean;
+  };
+  cdn: {
+    ingestionInfo: {
+      /**
+       * streamName is actually a secret stream key
+       */
+      streamName: string;
+      ingestionAddress: string;
+      backupIngestionAddress: string;
+    };
+    resolution: string;
+    frameRate: string;
+  };
```

Comment:
thanks for the documentation, but i'm wondering if we should be including every field here or just the ones that are relevant to us

Reply:
I tried to list only properties that we in some way use in our frontend or backend code.